### PR TITLE
Tidy up Chemistry-Holder a little.

### PR DESCRIPTION
**Pull request by TheTyrant25**
[INTERNAL]
## About the PR
Removes the single implicit type declaration of `/datum/reagents`. Also cleans up some missing doc comments. Doesn't add any new info, just rearranges.
## Why's this needed?
Code quality is a nice thing to have. Might as well do it while I'm here.

### CODE REVIEW

File: code/modules/chemistry/Chemistry-Holder.dm [hunk 1]
```diff
@@ -22,641 +22,696 @@ proc/chem_helmet_check(mob/living/carbon/human/H, var/what_liquid="hot")
...
+	var/composite_heat_capacity = 0
 
-		var/defer_reactions = 0 //Set internally to prevent reactions inside reactions.
-		var/deferred_reaction_checks = 0
-		var/processing_reactions = 0
-		var/inert = 0 //Do not react. At all. Do not pass go, do not collect $200. Halt. Stop right there, son.
+	var/defer_reactions = 0 ///Set internally to prevent reactions inside reactions.
+	var/deferred_reaction_checks = 0
+	var/processing_reactions = 0
+	var/inert = 0 ///Do not react. At all. Do not pass go, do not collect $200. Halt. Stop right there, son.
```
Comment (ZeWaka) [inline code/modules/chemistry/Chemistry-Holder.dm:39]: not how you do doc comments

File: code/modules/chemistry/Chemistry-Holder.dm [hunk 2]
```diff
@@ -22,641 +22,696 @@ proc/chem_helmet_check(mob/living/carbon/human/H, var/what_liquid="hot")
...
 
-		var/temperature_cap = 10000
-		var/temperature_min = 0
+	var/temperature_cap = 10000
+	var/temperature_min = 0
 
-		var/postfoam = 0 //attempt at killing infinite foam
-		var/can_be_heated = TRUE //can be heated by external sources
+	var/postfoam = 0 ///attempt at killing infinite foam
+	var/can_be_heated = TRUE ///can be heated by external sources
```
Comment (ZeWaka) [inline code/modules/chemistry/Chemistry-Holder.dm:53]: x

File: code/modules/chemistry/Chemistry-Holder.dm [hunk 3]
```diff
@@ -22,641 +22,696 @@ proc/chem_helmet_check(mob/living/carbon/human/H, var/what_liquid="hot")
...
+		amount = min(amount, target_reagents.maximum_volume - target_reagents.total_volume)
+		if(amount <= 0) return
+
+		if (do_fluid_react && issimulatedturf(target))
+			var/turf/simulated/T = target
+			return T.fluid_react(src, amount, index = index)
+
+		return trans_to_direct(target_reagents, amount, multiplier, index = index)
+
+	/// MBC note : I added update_target_reagents and update_self_reagents vars for fluid handling. y'see, there are a ton of transfer operations involving fluids that don't need to update reagents immediately as they happen.
```
Comment (ZeWaka) [inline code/modules/chemistry/Chemistry-Holder.dm:309]: this is not a doc comment.

File: code/modules/chemistry/Chemistry-Holder.dm [hunk 4]
```diff
@@ -665,512 +720,458 @@ datum
...
+				if (prob(5))
+					shock = 1
+			if ("bump")
+				shock = rand(3, 8)
+			if ("pushdown")
+				shock = rand(8, 16)
+		if (shock)
+			physical_shock(shock)
+
+	/// there were two different implementations, one of which didn't work, so i moved the working one here and both call it now - IM
```
Comment (ZeWaka) [inline code/modules/chemistry/Chemistry-Holder.dm:1130]: not a doc comment...

File: code/modules/chemistry/Chemistry-Holder.dm [hunk 5]
```diff
@@ -665,512 +720,458 @@ datum
...
+
+	proc/get_average_rgb()
+		var/datum/color/average = get_average_color()
+		return rgb(average.r, average.g, average.b)
+
+	/// give a list of the n tastes that are most present in this mixture
+	/// takes argument of how many tastes
+	proc/get_prevalent_tastes(var/num_val)
+		RETURN_TYPE(/list)
+		/// associative list with key = taste, val = total volume
```
Comment (ZeWaka) [inline code/modules/chemistry/Chemistry-Holder.dm:1027]: not how doc comments work

**Review by ZeWaka — CHANGES_REQUESTED** [review]
please review all your changed comments

**Comment by TheTyrant25** [comment]
should be all ready to go now

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**